address.go

```diff
@@ -83,6 +81,9 @@ type Address interface {
 	// Type returns the type of the address.
 	Type() AddressType
 
+	// ID returns the address ID.
+	ID() []byte
```

Comment:
```suggestion
    // ID returns the address ID, which is the concatenation of type prefix and raw address.
	ID() []byte
```

Something more helpful here would be great, because "ID" is super generic.

Do we still need a dedicated `Key` function in the interface? Can a caller not just use `string(addr.ID())` instead? 

Edit: I saw the Key used a bunch of times so perhaps it's convenient enough to keep it. Your call.